data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropMessage.avsc

```diff
@@ -0,0 +1,22 @@
+{
+  "type": "record",
+  "name": "InteropMessage",
+  "namespace": "net.corda.data.interop",
+  "fields": [
+      {
+        "doc": "Unique message identifier.",
+        "name": "messageId",
+        "type": "string"
+     },
+    {
+      "doc": "Request contains json data of facade request.",
+      "name": "request",
+      "type": "string"
+    },
+    {
+      "doc": "Response contains json data of facade response.",
```

Comment:
This schema suggests that you have both request and response in a single message, which is wrong. We could make both fields nullable, but we couldn't impose only one to be present.
What about changing this to:  
```
{
       "fields": [
               "type": "enum",  "name": "request", value =Request/response
              name: "payload",  type: String,
              doc: ""
              ]
 }
 ```
 or anything to make only one filed obligatory.

Reply:
I agree with your first point but I think its not good representation of schema adding request enum with value request or response. I personally feel its better to keep it simple as 

> {
      "doc": "Payload contains json data of facade request.",
      "name": "payload",
      "type": "string"
    }

here payload can be request or response based on consumer. Another way I am thinking, we can have separate schema for response based on how it evolve. We may need some additional fields while returning the response.

Whats your opinion?